src/pymor/reductors/mt.py

```diff
@@ -0,0 +1,203 @@
+            A = to_matrix(fom.A, format='dense')
+            E = None if isinstance(fom.E, IdentityOperator) else to_matrix(
+                fom.E, format='dense')
+
+            if symmetric:
+                poles, ev_r = spla.eigh(A, E)
+                rev = fom.A.source.from_numpy(ev_r.T)
+                lev = rev.copy()
+            else:
+                poles, ev_l, ev_r = spla.eig(A, E, left=True)
+                rev = fom.A.source.from_numpy(ev_r.T)
+                lev = fom.A.source.from_numpy(ev_l.T)
+            if which == 'SM':
+                dominance = np.abs(poles)
+            elif which == 'LR':
+                dominance = -poles.real
+            else:
+                absres = np.empty(len(poles))
+                for i in range(len(poles)):
+                    lev[i].scal(1 / lev[i].inner(fom.E.apply(rev[i]))[0][0])
+                    b_norm = fom.B.apply_adjoint(lev[i]).norm()[0]
+                    c_norm = fom.C.apply(rev[i]).norm()[0]
+                    absres[i] = b_norm * c_norm
+                if which == 'NR':
+                    dominance = -(absres / np.abs(np.real(poles)))
+                elif which == 'NS':
+                    dominance = -(absres / np.abs(poles))
+                elif which == 'NM':
+                    dominance = -np.array(absres)
+        elif decomposition == 'samdp':
+            poles, res, rev, lev = samdp(fom.A, fom.E,
+                                         fom.B.as_range_array(),
+                                         fom.C.as_source_array(),
+                                         r, **method_options)
+            absres = spla.norm(res, axis=(1, 2), ord=2)
+            if method_options['which'] == 'LR':
+                dominance = -(absres / np.abs(np.real(poles)))
+            elif method_options['which'] == 'LS':
+                dominance = -(absres / np.abs(poles))
+            elif method_options['which'] == 'LM':
+                dominance = -np.array(absres)
+
+        idx = sorted(range(len(poles)),
+                     key=lambda i: (dominance[i], -poles[i].real,
+                                    abs(poles[i].imag), 0 if poles[i].imag >= 0 else 1))
+        poles = poles[idx]
+        rev = rev[idx]
+        lev = lev[idx]
+        poles = poles[:r]
+        lev = lev[:r]
+        rev = rev[:r]
+
+        self.V = fom.A.source.empty(reserve=r)
+        self.W = fom.A.source.empty(reserve=r)
+
+        real_index = np.where(np.abs(poles.imag) / np.abs(poles) < 1e-6)[0]
+        complex_index = np.where((np.abs(poles.imag) / np.abs(poles) >= 1e-6) & (poles.imag > 0))[0]
+
+        if complex_index.size > 0 and complex_index[-1] == r-1:
+            self.logger.warning(
+                'Chosen order r will split complex conjugated pair of poles.')
+            if allow_complex_rom:
+                self.logger.info('Reduced model will be complex.')
+                self.V.append(rev[r-1])
+                self.W.append(lev[r-1])
```

Comment:
Using `-1` to index the last element is a bit nicer.
Similar below, in the `else` branch.